vdirsyncer/storage/base.py

```diff
@@ -92,6 +92,7 @@ class to obtain a storage instance pointing to this collection. It
             from the last segment of a URL or filesystem path.
 
         """
+        yield  # Needs to be an async generator
```

Comment:
```suggestion
        if false:
            yield  # Needs to be an async generator
```

Reply:
Ah, yes, that's better.

Reply:
You can also put the yield after the raise

Reply:
I expect mypy would complain that it's unreachable code.

I'm surprised flake doesn't pick that up.